File: Sofra.API/Entities/DiningTable.cs
using Sofra.API.Enums;

namespace Sofra.API.Entities;

public class DiningTable
{
    public int Id { get; set; }
    public int Number { get; set; }
    public int Capacity { get; set; }
    public int ZoneId { get; set; }
    public Zone Zone { get; set; } = null!;
    public int TableTypeId { get; set; }
    public TableType TableType { get; set; } = null!;
    public TableStatus Status { get; set; }
    public int? WaiterId { get; set; }
    public ApplicationUser? Waiter { get; set; }
    public string QrCode { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
}

File: sofra_desktop/lib/models/dining_table.dart
import 'table_status.dart';

class DiningTable {
  const DiningTable({
    required this.id,
    required this.number,
    required this.capacity,
    required this.zoneId,
    required this.zoneName,
    required this.tableTypeId,
    required this.tableTypeName,
    required this.status,
    required this.qrCode,
    required this.isActive,
    this.waiterId,
    this.waiterName,
  });

  final int id;
  final int number;
  final int capacity;
  final int zoneId;
  final String zoneName;
  final int tableTypeId;
  final String tableTypeName;
  final TableStatus status;
  final int? waiterId;
  final String? waiterName;
  final String qrCode;
  final bool isActive;

  factory DiningTable.fromJson(Map<String, dynamic> json) => DiningTable(
        id: json['id'] as int,
        number: json['number'] as int,
        capacity: json['capacity'] as int,
        zoneId: json['zoneId'] as int,
        zoneName: json['zoneName'] as String,
        tableTypeId: json['tableTypeId'] as int,
        tableTypeName: json['tableTypeName'] as String,
        status: TableStatus.fromValue(json['status'] as int),
        waiterId: json['waiterId'] as int?,
        waiterName: json['waiterName'] as String?,
        qrCode: json['qrCode'] as String,
        isActive: json['isActive'] as bool,
      );
}
